systemtest/src/test/java/io/strimzi/systemtest/metrics/StrimziMetricsReporterST.java

```diff
@@ -110,12 +179,35 @@ void setupEnvironment() {
                             .endValues()
                         .endStrimziMetricsReporterConfig()
                     .endKafka()
-                .endSpec().build()
+                .endSpec().build(),
+            KafkaTemplates.kafkaWithoutEntityOperator(testStorage.getNamespaceName(), testStorage.getClusterName() + "-tgt", 1).build()
         );
 
         KubeResourceManager.get().createResourceWithWait(
+            KafkaConnectTemplates.kafkaConnectWithFilePlugin(testStorage.getNamespaceName(), "my-connect",
+                    testStorage.getClusterName(), 1)
+                .editOrNewSpec()
+                    .withNewStrimziMetricsReporterConfig()
+                        .withNewValues()
+                            .withAllowList("kafka_connect.*")
+                        .endValues()
+                    .endStrimziMetricsReporterConfig()
+                .endSpec().build(),
+            KafkaMirrorMaker2Templates.kafkaMirrorMaker2(testStorage.getNamespaceName(), "my-mm2",
```

Comment:
Same here.

Reply:
We are using it everywhere else:

`KafkaMirrorMaker2Templates.kafkaMirrorMaker2(testStorage.getNamespaceName(), testStorage.getClusterName(), testStorage.getSourceClusterName(), testStorage.getTargetClusterName(), 1, false)`

Why it would not work here as well?

Reply:
with your suggested change, we are creating both KC and MM2 with the same name, and the role binding reconciliation logic is in common between these components

Reply:
Ah okay, I understand.
So anyway, instead of hardcoding it, it would be better to create some local variables for the class (similarly how it is done in `MetricsST`) for better manipulation in other tests.

Also, I'm thinking about leaving the creation of the MM2 and Connect (and possibly other components in the future) for the particular tests.
So in case that one test (for Kafka for example) fails, the suite will not try to recreate MM2 and Connect.

That's I guess the right way (plus for example in the `MetricsST` we can move the second Kafka cluster out of the before all, as there is no MM1 anymore) - and it will speed things up. WDYT?

Reply:
> in case that one test (for Kafka for example) fails, the suite will not try to recreate MM2 and Connect

wdym here? all tests are always executed even if one ore more fail

Reply:
Isn't the issue because you have different namespaces in beforeall and in the test?

Reply:
I'm always using `testStorage.getNamespaceName()`, and looking at logs it's the same namespace (test-suite-namespace).
[Uploading 2025-06-24-10-44-07.zip…]()

Reply:
Yes, I'm doing exactly that, but still running some tests to confirm is all good.